no type named 'ExternalAsciiStringResource' in 'v8::String'

I can't install term2 on my new mac,
Atom v0.177.0
OS X 10.10.2
XCode 6.1.1(6A2008a)
```
> pty.js@0.2.5 install /private/var/folders/05/lbfncb6x4_s1gxryqnr6yyn80000gn/T/apm-install-dir-115110-700-poea8n/node_modules/term2/node_modules/pty.js
> node-gyp rebuild

  CXX(target) Release/obj.target/pty/src/unix/pty.o

In file included from ../src/unix/pty.cc:16:
../node_modules/nan/nan.h:67:21: error: no type named 'ExternalAsciiStringResource' in 'v8::String'; did you mean 'ExternalStringResource'?
typedef v8::String::ExternalAsciiStringResource
        ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
                    ExternalStringResource
/Users/Learning/.atom/.node-gyp/.node-gyp/0.21.0/deps/v8/include/v8.h:1936:19: note: 'ExternalStringResource' declared here
  class V8_EXPORT ExternalStringResource
                  ^
In file included from ../src/unix/pty.cc:16:
In file included from ../node_modules/nan/nan.h:74:
../node_modules/nan/nan_new.h:164:26: error: class member cannot be redeclared
  static inline return_t New(NanExternalOneByteStringResource * value);
                         ^
../node_modules/nan/nan_new.h:163:26: note: previous declaration is here
  static inline return_t New(v8::String::ExternalStringResource * value);
                         ^
In file included from ../src/unix/pty.cc:16:
In file included from ../node_modules/nan/nan.h:74:
In file included from ../node_modules/nan/nan_new.h:190:
../node_modules/nan/nan_implementation_12_inl.h:222:22: error: redefinition of 'New'
Factory<v8::String>::New(NanExternalOneByteStringResource * value) {
                     ^
../node_modules/nan/nan_implementation_12_inl.h:217:22: note: previous definition is here
Factory<v8::String>::New(v8::String::ExternalStringResource * value) {
                     ^
In file included from ../src/unix/pty.cc:16:
In file included from ../node_modules/nan/nan.h:74:
../node_modules/nan/nan_new.h:319:1: error: redefinition of 'NanNew'
NanNew(NanExternalOneByteStringResource * value) {
^
../node_modules/nan/nan_new.h:313:1: note: previous definition is here
NanNew(v8::String::ExternalStringResource * value) {
^
In file included from ../src/unix/pty.cc:16:
../node_modules/nan/nan.h:1790:17: error: no member named 'GetExternalAsciiStringResource' in 'v8::String'; did you mean 'GetExternalOneByteStringResource'?
    return str->GetExternalAsciiStringResource();
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                GetExternalOneByteStringResource
/Users/Learning/.atom/.node-gyp/.node-gyp/0.21.0/deps/v8/include/v8.h:2003:40: note: 'GetExternalOneByteStringResource' declared here
  const ExternalOneByteStringResource* GetExternalOneByteStringResource() const;
                                       ^
In file included from ../src/unix/pty.cc:16:
../node_modules/nan/nan.h:1790:12: error: cannot initialize return object of type 'const NanExternalOneByteStringResource *' (aka 'const v8::String::ExternalStringResource *') with an rvalue of type 'const v8::String::ExternalOneByteStringResource *'
    return str->GetExternalAsciiStringResource();
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:1800:17: error: no member named 'IsExternalAscii' in 'v8::String'; did you mean 'IsExternal'?
    return str->IsExternalAscii();
                ^~~~~~~~~~~~~~~
                IsExternal
/Users/Learning/.atom/.node-gyp/.node-gyp/0.21.0/deps/v8/include/v8.h:1900:8: note: 'IsExternal' declared here
  bool IsExternal() const;
       ^
In file included from ../src/unix/pty.cc:16:
../node_modules/nan/nan.h:1825:11: error: assigning to 'const char *' from incompatible type 'const uint16_t *' (aka 'const unsigned short *')
    *data = ext->data();
          ^ ~~~~~~~~~~~
../node_modules/nan/nan.h:1868:10: error: no matching function for call to 'Encode'
  return node::Encode(
         ^~~~~~~~~~~~
/Users/Learning/.atom/.node-gyp/.node-gyp/0.21.0/src/node.h:272:34: note: candidate function not viable: cannot convert argument of incomplete type 'const void *' to 'const char *'
NODE_EXTERN v8::Local<v8::Value> Encode(v8::Isolate* isolate,
                                 ^
/Users/Learning/.atom/.node-gyp/.node-gyp/0.21.0/src/node.h:278:34: note: candidate function not viable: requires 3 arguments, but 4 were provided
NODE_EXTERN v8::Local<v8::Value> Encode(v8::Isolate* isolate,
                                 ^
/Users/Learning/.atom/.node-gyp/.node-gyp/0.21.0/src/node.h:283:45: note: candidate function not viable: requires at most 3 arguments, but 4 were provided
                inline v8::Local<v8::Value> Encode(
                                            ^
/Users/Learning/.atom/.node-gyp/.node-gyp/0.21.0/src/node.h:64:53: note: expanded from macro 'NODE_DEPRECATED'
#define NODE_DEPRECATED(msg, fn) V8_DEPRECATED(msg, fn)
                                                    ^
/Users/Learning/.atom/.node-gyp/.node-gyp/0.21.0/deps/v8/include/v8config.h:329:45: note: expanded from macro 'V8_DEPRECATED'
# define V8_DEPRECATED(message, declarator) declarator
                                            ^
9 errors generated.
make: *** [Release/obj.target/pty/src/unix/pty.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:12)
gyp ERR! System Darwin 14.1.0
gyp ERR! command "node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /private/var/folders/05/lbfncb6x4_s1gxryqnr6yyn80000gn/T/apm-install-dir-115110-700-poea8n/node_modules/term2/node_modules/pty.js
gyp ERR! node -v v0.10.35
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok 
npm ERR! Darwin 14.1.0
npm ERR! argv "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Applications/Atom.app/Contents/Resources/app/apm/.apmrc" "--userconfig" "/Users/Learning/.atom/.apmrc" "install" "/private/var/folders/05/lbfncb6x4_s1gxryqnr6yyn80000gn/T/d-115110-700-kxmpt1/package.tgz" "--target=0.21.0" "--arch=x64"
npm ERR! node v0.10.35
npm ERR! npm  v2.3.0
npm ERR! code ELIFECYCLE

npm ERR! pty.js@0.2.5 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the pty.js@0.2.5 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the pty.js package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls pty.js
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/var/folders/05/lbfncb6x4_s1gxryqnr6yyn80000gn/T/apm-install-dir-115110-700-poea8n/npm-debug.log
```

--- a/index.coffee
+++ b/index.coffee
@@ -43,20 +43,22 @@ module.exports =
     activate: (@state)->
 
       ['up', 'right', 'down', 'left'].forEach (direction)=>
-        atom.workspaceView.command "term2:open-split-#{direction}", @splitTerm.bind(this, direction)
+        atom.commands.add "atom-workspace", "term2:open-split-#{direction}", @splitTerm.bind(this, direction)
 
-      atom.workspaceView.command "term2:open", @newTerm.bind(this)
-      atom.workspaceView.command "term2:pipe-path", @pipeTerm.bind(this, 'path')
-      atom.workspaceView.command "term2:pipe-selection", @pipeTerm.bind(this, 'selection')
+      atom.commands.add "atom-workspace", "term2:open", @newTerm.bind(this)
+      atom.commands.add "atom-workspace", "term2:pipe-path", @pipeTerm.bind(this, 'path')
+      atom.commands.add "atom-workspace", "term2:pipe-selection", @pipeTerm.bind(this, 'selection')
 
     getColors: ->
-      {colors: {
+      console.log atom.config.get 'term2.colors.normalBlack'
+      console.log atom.config.getAll 'term2.colors'
+      {
         normalBlack, normalRed, normalGreen, normalYellow
         normalBlue, normalPurple, normalCyan, normalWhite
         brightBlack, brightRed, brightGreen, brightYellow
         brightBlue, brightPurple, brightCyan, brightWhite
         background, foreground
-      }} = atom.config.getSettings().term2
+      } = (atom.config.getAll 'term2.colors')[0].value
       [
         normalBlack, normalRed, normalGreen, normalYellow
         normalBlue, normalPurple, normalCyan, normalWhite
@@ -110,7 +112,7 @@ module.exports =
       pane.activateItem item
 
     pipeTerm: (action)->
-      editor = atom.workspace.getActiveEditor()
+      editor = @getActiveEditor()
       stream = switch action
         when 'path'
           editor.getBuffer().file.path

--- a/lib/TermView.coffee
+++ b/lib/TermView.coffee
@@ -8,7 +8,8 @@ Terminal   = require 'atom-term.js'
 
 keypather  = do require 'keypather'
 
-{$, View, Task} = require 'atom'
+{Task} = require 'atom'
+{$, View} = require 'atom-space-pen-views'
 
 last = (str)-> str[str.length-1]
 
@@ -29,12 +30,12 @@ class TermView extends View
     opts.shellArguments or= ''
 
     editorPath = keypather.get atom, 'workspace.getEditorViews[0].getEditor().getPath()'
-    opts.cwd = opts.cwd or atom.project.getPath() or editorPath or process.env.HOME
+    opts.cwd = opts.cwd or atom.project.getPaths()[0] or editorPath or process.env.HOME
     super
 
   forkPtyProcess: (args=[])->
     processPath = require.resolve './pty'
-    path = atom.project.getPath() ? '~'
+    path = atom.project.getPaths()[0] ? '~'
     Task.once processPath, fs.absolute(path), args
 
   initialize: (@state)->
@@ -85,8 +86,8 @@ class TermView extends View
   attachEvents: ->
     @resizeToPane = @resizeToPane.bind this
     @attachResizeEvents()
-    @command "term2:paste", => @paste()
-    @command "term2:copy", => @copy()
+    atom.commands.add "atom-workspace", "term2:paste", => @paste()
+    atom.commands.add "atom-workspace", "term2:copy", => @copy()
 
   copy: ->
     if  @term._selected  # term.js visual mode selections
@@ -131,7 +132,7 @@ class TermView extends View
 
     @resize cols, rows
     @term.resize cols, rows
-    atom.workspaceView.getActivePaneView().css overflow: 'visible'
+    atom.views.getView(atom.workspace).style.overflow = 'visible'
 
   getDimensions: ->
     fakeCol = $("<span id='colSize'>&nbsp;</span>").css visibility: 'hidden'

--- a/package.json
+++ b/package.json
@@ -17,6 +17,7 @@
   },
   "dependencies": {
     "atom-term.js": "0.0.55",
+    "atom-space-pen-views": "^2.0.3",
     "debounce": "0.0.3",
     "fs-plus": "^2.2.3",
     "keypather": "^1.3.2",

